### PR TITLE
feat(sim): socket simulators use hostname instead of ip

### PR DESCRIPTION
**Pull request by amitlissack**
Very stubbornly not using boost for some reason.

### CODE REVIEW

**Review by sfoster1 — APPROVED** [review]
This looks pretty good, couple quibbles with unistd api handling

File: include/can/simlib/socket_transport.hpp [hunk 1]
```diff
@@ -115,4 +118,15 @@ auto SocketTransport<CriticalSection>::read(uint32_t &arb_id, uint8_t *buff,
     return true;
 }
 
+template <synchronization::LockableProtocol CriticalSection>
+auto SocketTransport<CriticalSection>::host_to_ip(const std::string &host)
+    -> std::string {
+    auto h = gethostbyname(host.c_str());
```
[the remark this answers is not in the document]
Comment (amitlissack) [reply include/can/simlib/socket_transport.hpp:124]: I am drawing a line in the sand. One more line for this code and I switch to boost. This is getting dumb.

File: include/can/simlib/socket_transport.hpp [hunk 2]
```diff
@@ -1,6 +1,7 @@
 #pragma once
 
 #include <arpa/inet.h>
+#include <netdb.h>
```
[the remark this answers is not in the document]
Comment (amitlissack) [reply include/can/simlib/socket_transport.hpp:4]: See my boost comment.

**Comment by DerekMaggio** [comment]
Going to test this right now. Thank you @amitlissack

**Comment by Laura-Danielle** [comment]
Why would boost be easier for something like this?

**Comment by DerekMaggio** [comment]
IT'S ALIIIIIIIIIIIIIIIIIIIIIVE!!!!
Thank you @amitlissack

**Review by DerekMaggio — APPROVED** [review]

**Comment by sfoster1** [comment]
> Why would boost be easier for something like this?

boost isn't necessarily easier to do an extremely straightforward, posix compat only, process like we have here of looking up a hostname and opening a socket to it and just bailing if it fails. if your requirements look more like "on a cross-platform basis, find a hostname and port and open a socket and handle various failures" it starts getting a little closer to the concision and flexibility of something like python's sockets library.

**Pull request merged**